src/sidebar/services/dashboard.ts

```diff
@@ -13,15 +13,39 @@ export class DashboardService {
     this._dashboardConfig = settings.dashboard;
   }
 
-  open() {
+  async getAuthToken(): Promise<string | undefined> {
     if (!this._rpc || !this._dashboardConfig) {
-      return;
+      return undefined;
     }
 
-    postMessageJsonRpc.notify(
+    return postMessageJsonRpc.call<string>(
       this._rpc.targetFrame,
       this._rpc.origin,
-      this._dashboardConfig.entryPointRPCMethod,
+      this._dashboardConfig.authTokenRPCMethod,
     );
   }
+
+  open(authToken: string, document_ = document) {
+    // The auth token is fetched separately, so that we can invoke this method
+    // as part of a user gesture and browsers don't block the new tab opened by
+    // the form being submitted later
+    if (!this._rpc || !this._dashboardConfig) {
+      return;
```

Comment:
If we get here then something has gone wrong. Probably worth logging something, or throwing an error.